Lychee link checker scans .direnv directory causing false errors

## Problem

The `lychee` markdown link checker (used by `cargo make check-markdown-links`) scans files inside the `.direnv/flake-inputs/` directory, which contains vendored Nix flake sources. This produces false errors for files that are not part of our repository:

```
[ERROR] file:///.../.direnv/flake-inputs/.../CONTRIBUTING.md#dont-be-an-as
[ERROR] file:///.../.direnv/flake-inputs/.../TEE.md
[ERROR] file:///.../.direnv/flake-inputs/.../tee_launcher/Makefile
[ERROR] file:///.../.direnv/flake-inputs/.../tee_launcher/deployment/build-image.sh
```

## Fix

Add `.direnv` to the `exclude_path` list in `lychee.toml`, similar to how `libs/nearcore`, `target`, and `.git` are already excluded.

--- a/lychee.toml
+++ b/lychee.toml
@@ -9,4 +9,4 @@ offline = true
 # Check anchor/fragment links (e.g., #section-name)
 include_fragments = true
 
-exclude_path = ["libs/nearcore", "target", ".git"]
+exclude_path = ["libs/nearcore", "target", ".git", ".direnv"]

